runtime-modules/proposals/engine/src/lib.rs

```diff
@@ -898,16 +845,101 @@ type ApprovedProposal<T> = ApprovedProposalData<
     <T as Trait>::ProposalId,
     <T as system::Trait>::BlockNumber,
     MemberId<T>,
-    types::BalanceOf<T>,
-    <T as stake::Trait>::StakeId,
+    BalanceOf<T>,
     <T as system::Trait>::AccountId,
 >;
 
 // Simplification of the 'Proposal' type
 type ProposalOf<T> = Proposal<
     <T as system::Trait>::BlockNumber,
     MemberId<T>,
-    types::BalanceOf<T>,
-    <T as stake::Trait>::StakeId,
+    BalanceOf<T>,
     <T as system::Trait>::AccountId,
 >;
+
+pub struct StakingManager<T: Trait, LockId: Get<LockIdentifier>> {
+    trait_marker: PhantomData<T>,
+    lock_id_marker: PhantomData<LockId>,
+}
+
+impl<T: Trait, LockId: Get<LockIdentifier>> StakingHandler<T> for StakingManager<T, LockId> {
```

Comment:
Why do we need to implement this again and again for each module, that seems wrong.

Reply:
It is situated here until we integrate at least 2 modules. I don't expect copies with this handler.

Reply:
OK, merging in that case.

Reply:
Ok.